File: app/src/main/java/com/mariotti/developer/futureclock/broadcast/StartupReceiver.java
package com.mariotti.developer.futureclock.broadcast;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class StartupReceiver extends BroadcastReceiver {
    private static final String TAG = "StartupReceiver";

    public StartupReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        /*Alarm alarm = UIAlarmController.getAlarmController(context)
                .getNextAlarm();
        if (alarm != null) {
            AlarmFiredActivity.setActivityAlarm(context, alarm.getUUID());
            Log.d(TAG, "Alarm set");
        }*/
    }
}

File: app/src/main/java/com/mariotti/developer/futureclock/broadcast/StartupReceiver.kt
package com.mariotti.developer.futureclock.broadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class StartupReceiver : BroadcastReceiver() {

    companion object {
        private val TAG = "StartupReceiver"
    }

    override fun onReceive(context: Context, intent: Intent) {
//        RxDatabaseAlarmController.getInstance(context)
        //                .getActiveAlarms()
        //                .observeOn(AndroidSchedulers.mainThread())
        //                .subscribe {
        //                    Log.d(TAG, "StartupReceiver")
        //                    if (it.size > 0) {
        //                        val alarmToFire: Alarm? = getNextAlarm(it)
        //                        alarmToFire?.let {
        //                            makeNotificationFromAlarm(context, it)
        //
        //                            // TODO - to uncomment when FiredAlarmFragment is completed
        //                            //FiredAlarmActivity.setActivityAlarm(context, it.uuid)
        //                        }
        //                    }
        //                }
    }
}
